votor/src/certificate_pool_service.rs

```diff
@@ -206,10 +212,19 @@ impl CertificatePoolService {
 
         // Ingest votes into certificate pool and notify voting loop of new events
         while !ctx.exit.load(Ordering::Relaxed) {
-            // TODO: we need set identity here as well
+            // Update the current pubkey if it has changed
+            {
+                let new_pubkey_guard = ctx.my_pubkey.read().unwrap();
```

Comment:
if we're going to grab a lock i'd prefer to just pass cluster_info here instead of adding a new one. Otherwise we can have another receiver for set identity events to check in the select below

Reply:
Adding an event receiver is probably over kill since this event is pretty rare. I considered passing cluster_info, the problem is that we are going to have a race between event loop resetting vote_history and certificate pool setting identity.

But I don't feel strongly about it, if you are sure the race will not do any harm then I can change to cluster_info.

Reply:
It should be no difference as this implementation can race anyway unless you hold the lock for the entire ingestion process.

If you change pubkey from staked to unstaked:
- you might still emit events based on the staked nodes votes, however you are unable to sign with the unstaked key

If you change pubkey from unstaked to staked:
- The unstaked node could not have signed any votes in the first place, so technically safeto* cannot trigger in the first place

Changing between two staked nodes is not supported (1 node_pubkey per vote account).

So I think just checking cluster_info is fine.